syntax = "proto3";

package semantic_router;

option go_package = "github.com/alibaba/higress/plugins/wasm-go/extensions/semantic-router/proto";

// 分类请求消息
message ClassificationRequest {
  // 待分类的文本内容
  string text = 1;
  // 请求唯一标识符
  string request_id = 2;
  // 是否启用缓存
  bool cache_enabled = 3;
  // 扩展元数据
  map<string, string> metadata = 4;
  // 超时时间（毫秒）
  int32 timeout_ms = 5;
}

// 分类响应消息
message ClassificationResponse {
  // 预测的类别
  string category = 1;
  // 置信度 (0.0 - 1.0)
  double confidence = 2;
  // 所有类别的得分
  map<string, double> scores = 3;
  // 推理延迟（毫秒）
  double latency_ms = 4;
  // 推理方法 ("model", "cache", "fallback")
  string method = 5;
  // 模型版本
  string model_version = 6;
  // 响应时间戳
  int64 timestamp = 7;
  // 缓存键（用于调试）
  string cache_key = 8;
}

// 批量分类请求
message BatchClassificationRequest {
  repeated ClassificationRequest requests = 1;
}

// 批量分类响应
message BatchClassificationResponse {
  repeated ClassificationResponse responses = 1;
}

// 健康检查请求
message HealthCheckRequest {
  // 检查类型 ("basic", "detailed")
  string check_type = 1;
}

// 健康检查响应
message HealthCheckResponse {
  // 服务状态 ("healthy", "unhealthy", "degraded")
  string status = 1;
  // 统计信息
  map<string, string> stats = 2;
  // 检查时间戳
  int64 timestamp = 3;
  // 详细信息
  string details = 4;
}

// 语义分类服务定义
service SemanticClassifierService {
  // 单个文本分类
  rpc Classify(ClassificationRequest) returns (ClassificationResponse);
  
  // 批量文本分类
  rpc BatchClassify(BatchClassificationRequest) returns (BatchClassificationResponse);
  
  // 流式分类（用于实时处理）
  rpc StreamClassify(stream ClassificationRequest) returns (stream ClassificationResponse);
  
  // 健康检查
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}